test/unit/logic/referral.spec.ts

```diff
@@ -254,6 +391,117 @@ describe('referral-component', () => {
       })
     })
 
+    describe('when referral reaches a tier milestone', () => {
+      describe.each([
+        { invitedUsers: 5, tier: 1, description: 'tier 1 milestone' },
+        { invitedUsers: 10, tier: 2, description: 'tier 2 milestone' },
+        { invitedUsers: 20, tier: 3, description: 'tier 3 milestone' },
+        { invitedUsers: 25, tier: 4, description: 'tier 4 milestone' },
+        { invitedUsers: 30, tier: 5, description: 'tier 5 milestone' },
+        { invitedUsers: 50, tier: 6, description: 'tier 6 milestone' },
+        { invitedUsers: 60, tier: 7, description: 'tier 7 milestone' },
+        { invitedUsers: 75, tier: 8, description: 'tier 8 milestone' },
+        { invitedUsers: 100, tier: 9, description: 'tier 9 milestone' }
+      ])('and the referral reaches $description', ({ invitedUsers, tier }) => {
+        beforeEach(() => {
+          mockReferralDb.findReferralProgress.mockResolvedValueOnce([
+            {
+              referrer: validReferrer,
+              invited_user: validInvitedUser,
+              status: ReferralProgressStatus.SIGNED_UP
+            }
+          ])
+          mockReferralDb.updateReferralProgress.mockResolvedValueOnce(undefined)
+          mockReferralDb.countAcceptedInvitesByReferrer.mockResolvedValueOnce(invitedUsers)
+        })
+
+        it(`should calculate correct tier for ${invitedUsers} invited users`, async () => {
+          await referralComponent.finalizeReferral(validInvitedUser)
+
+          expect(mockSns.publishMessage).toHaveBeenCalledWith(
+            expect.objectContaining({
+              metadata: expect.objectContaining({
+                tier,
+                invitedUsers
+              })
+            })
+          )
+        })
+      })
+    })
+
+    describe('when referral does not reach a tier milestone', () => {
+      describe.each([
+        { invitedUsers: 1, tier: 1 },
+        { invitedUsers: 3, tier: 1 },
+        { invitedUsers: 7, tier: 2 },
+        { invitedUsers: 15, tier: 3 },
+        { invitedUsers: 22, tier: 4 },
+        { invitedUsers: 28, tier: 5 },
+        { invitedUsers: 45, tier: 6 },
+        { invitedUsers: 55, tier: 7 },
+        { invitedUsers: 70, tier: 8 },
+        { invitedUsers: 95, tier: 9 }
+      ])('and the referral has $invitedUsers invited users', ({ invitedUsers, tier }) => {
+        beforeEach(() => {
+          mockReferralDb.findReferralProgress.mockResolvedValueOnce([
+            {
+              referrer: validReferrer,
+              invited_user: validInvitedUser,
+              status: ReferralProgressStatus.SIGNED_UP
+            }
+          ])
+          mockReferralDb.updateReferralProgress.mockResolvedValueOnce(undefined)
+          mockReferralDb.countAcceptedInvitesByReferrer.mockResolvedValueOnce(invitedUsers)
+        })
+
+        it(`should calculate correct tier for ${invitedUsers} invited users`, async () => {
```

Comment:
As these tests are expecting a notification to be sent, what do you think about changing the `it` so it reflects what's expected?
```suggestion
        it(`should send the notification with the correct tier for ${invitedUsers} invited users`, async () => {
```